Framework/Algorithms/src/EQSANSCorrectFrame.cpp

```diff
@@ -58,32 +65,41 @@ void EQSANSCorrectFrame::exec() {
   const double minTOF = getProperty("MinTOF");
   const double frameWidth = getProperty("FrameWidth");
   const bool isFrameSkipping = getProperty("FrameSkipping");
-  const auto &spectrumInfo = inputWS->spectrumInfo();
+  const bool pathToPixel = getProperty("PathToPixel");
+  const std::string detectorName = getProperty("DetectorName");
 
+  const auto &spectrumInfo = inputWS->spectrumInfo();
+  auto ins = inputWS->getInstrument();
+  auto sam = ins->getSample();
+  auto mod = ins->getSource();
+  const auto msd = mod->getDistance(*sam);
+  const auto det = ins->getComponentByName(detectorName);
+  const auto mdd = mod->getDistance(*det);
```

Comment:
This could be done with [`spectrumInfo.l2(...)`](https://docs.mantidproject.org/nightly/api/python/mantid/api/SpectrumInfo.html#mantid.api.SpectrumInfo.l2)

Reply:
Unfortunately I can't use `l2` because `ins->getComponentByName(detectorName)` represents the whole detector panel. As far as I know, `spectrumInfo.l2(...)` can only be applied to detector pixels. I use `spectrumInfo.l2(...)` later ([ line 132](https://github.com/mantidproject/mantid/blob/c179dc2694ea5e0aa5eb95ae0f8050933c7ec279/Framework/Algorithms/src/EQSANSCorrectFrame.cpp#L132)) when looking at particular detector pixels.